main.go

```diff
@@ -0,0 +1,227 @@
+	}
+
+	// // Initialize Tracing instance
+	// tracer, err := tracing.New(service.Name, service.TraceURL)
+	// if err != nil {
+	//      log.Err("Tracing Init Failed", err.Error())
+	//      os.Exit(1)
+	// }
+
+	// Initialize Handler intance
+	handler := cilium.New(cfg, log, kubeconfigHandler)
+	handler = adapter.AddLogger(log, handler)
+
+	service.Handler = handler
+	service.Channel = make(chan interface{}, 10)
+	service.StartedAt = time.Now()
+	service.Version = version
+	service.GitSHA = gitsha
+	go registerCapabilities(service.Port, log)        //Registering static capabilities
+	go registerDynamicCapabilities(service.Port, log) //Registering latest capabilities periodically
+
+	// Server Initialization
+	log.Info("Adaptor Listening at port: ", service.Port)
+	err = grpc.Start(service, nil)
+	if err != nil {
+		log.Error(err)
+		os.Exit(1)
+	}
+}
+
+func isDebug() bool {
+	return os.Getenv("DEBUG") == "true"
+}
+
+func mesheryServerAddress() string {
+	meshReg := os.Getenv("MESHERY_SERVER")
+
+	if meshReg != "" {
+		if strings.HasPrefix(meshReg, "http") {
+			return meshReg
+		}
+
+		return "http://" + meshReg
+	}
+
+	return "http://localhost:9081"
+}
+
+func serviceAddress() string {
+	svcAddr := os.Getenv("SERVICE_ADDR")
+
+	if svcAddr != "" {
+		return svcAddr
+	}
+
+	return "mesherylocal.layer5.io"
+}
+
+func registerCapabilities(port string, log logger.Handler) {
+	// Register workloads
+	log.Info("Registering static workloads...")
+	if err := oam.RegisterWorkloads(mesheryServerAddress(), serviceAddress()+":"+port); err != nil {
+		log.Info(err.Error())
+	}
+	log.Info("Registering static workloads completed")
+	// Register traits
+	if err := oam.RegisterTraits(mesheryServerAddress(), serviceAddress()+":"+port); err != nil {
+		log.Info(err.Error())
+	}
+}
+
+func registerDynamicCapabilities(port string, log logger.Handler) {
+	registerWorkloads(port, log)
+	//Start the ticker
+	const reRegisterAfter = 24
+	ticker := time.NewTicker(reRegisterAfter * time.Hour)
+	for {
+		<-ticker.C
+		registerWorkloads(port, log)
+	}
+
+}
+func registerWorkloads(port string, log logger.Handler) {
+	var url string
+	var gm string
+	// Prechecking to skip comp gen
+	release, err := config.GetLatestReleases(1)
+	if err != nil {
+		log.Info("Could not get latest stable release")
+		return
+	}
+	version := release[0].TagName
+	if os.Getenv("FORCE_DYNAMIC_REG") != "true" && oam.AvailableVersions[version] {
+		log.Info("Components available statically for version ", version, ". Skipping dynamic component registeration")
+		return
+	}
+	//If a URL is passed from env variable, it will be used for component generation with default method being "using manifests"
+	// In case a helm chart URL is passed, COMP_GEN_METHOD env variable should be set to Helm otherwise the component generation fails
+	if os.Getenv("COMP_GEN_URL") != "" {
+		url = os.Getenv("COMP_GEN_URL")
+		if os.Getenv("COMP_GEN_METHOD") == "Helm" || os.Getenv("COMP_GEN_METHOD") == "Manifest" {
```

Comment:
Is there a helm chart available, too?

Reply:
Yup, they can be found over [here](https://github.com/cilium/cilium/tree/master/install/kubernetes/cilium)